app/waf/src/models/as3.model.ts

```diff
@@ -249,7 +249,7 @@ export class AS3Action extends AS3Object {
       if (!isNullOrUndefined(this.wafpolicy)) {
         let wafid = <classwafpolicy>this.wafpolicy;
         for (let onewaf of this.wafs) {
-          if (onewaf.id === wafid.wafpolicy) {
+          if (onewaf.id === wafid.wafpolicy && onewaf.name) {
```

Comment:
onewaf -> waf?

Reply:
It will be removed by our new as3 translation code in next commit. Needn't to improve it any more.